### PR TITLE
[NEW] Print Transcript on Mobile Setting

### PR DESCRIPTION
**Pull request by AlexanderKanakis**
For issue: https://github.com/WideChat/Rocket.Chat/issues/1200
Related PRs: https://github.com/WideChat/Rocket.Chat.Livechat/pull/146

### CODE REVIEW

**Review by Shailesh351 — COMMENTED** [review]
@AlexanderKanakis The functionality is working as expected. 

I just observed one thing, `Livechat_enable_transcript` and `Livechat_enable_print_transcript` are different. The former is for the `Save Transcript` button and the latter is for the `Print Chat` option in the three-dot menu.

So naming this new setting `Livechat_enable_print_transcript_on_mobile` is kinda confusing. How about we change it to `Livechat_enable_transcript_mobile`?

**Comment by Shailesh351** [comment]
@ear-dev I understand it's not relevant to this PR but how about we update the Setting name `Ask Visitor if They Would Like a Transcript After Chat Closed` to `Enable Save Transcript` since we moved away from the prompt to button? Just to avoid any confusion in naming

**Comment by ear-dev** [comment]
> @ear-dev I understand it's not relevant to this PR but how about we update the Setting name `Ask Visitor if They Would Like a Transcript After Chat Closed` to `Enable Save Transcript` since we moved away from the prompt to button? Just to avoid any confusion in naming

I think that's a good idea.  Thanks Shailesh.  @AlexanderKanakis can you include that small fix, to the text next to the button, in this PR as well please?

**Comment by ear-dev** [comment]
@AlexanderKanakis I'm seeing this button twice now?  Are you also seeing that?

image

**Comment by AlexanderKanakis** [comment]
> @AlexanderKanakis I'm seeing this button twice now? Are you also seeing that?
> 
> image

@ear-dev Yup, that's the db not deleting older entries. If you have access to your db find the `Livechat_enable_print_transcript_on_mobile` entry in `rocketchat_settings` and delete it.

EDIT: In case you still see it, clear your cache.

**Comment by ear-dev** [comment]
@AlexanderKanakis gottit......  @Shailesh351 if this looks good can you approve when ready please?  thanks

File: app/livechat/server/config.ts [hunk 1]
```diff
@@ -552,6 +552,14 @@ Meteor.startup(function () {
 			i18nLabel: 'Transcript_Print_Enabled',
 		});
 
+		this.add('Livechat_enable_transcript_mobile', false, {
+			type: 'boolean',
+			group: 'Omnichannel',
+			public: true,
+			i18nLabel: 'Transcript_Print_Enabled_Mobile',
```
Comment (Shailesh351) [inline app/livechat/server/config.ts]: I think we should also rename this one as well.

File: app/livechat/server/methods/getInitialData.js [hunk 2]
```diff
@@ -89,6 +89,7 @@ Meteor.methods({
 		info.transcript = initSettings.Livechat_enable_transcript;
 		info.transcriptMessage = initSettings.Livechat_transcript_message;
 		info.printTranscript = initSettings.Livechat_enable_print_transcript;
+		info.printMobileTranscript = initSettings.Livechat_enable_transcript_mobile;
```
Comment (Shailesh351) [inline app/livechat/server/methods/getInitialData.js]: Also rename`printMobileTranscript`

**Review by Shailesh351 — COMMENTED** [review]
The package-lock.json file is having huge changes in this PR, but there are no changes in the package.json file? Can we remove it if not needed?

File: app/livechat/server/config.ts [hunk 3]
```diff
@@ -552,6 +552,14 @@ Meteor.startup(function () {
 			i18nLabel: 'Transcript_Print_Enabled',
 		});
 
+		this.add('Livechat_enable_transcript_mobile', false, {
+			type: 'boolean',
+			group: 'Omnichannel',
+			public: true,
+			i18nLabel: 'Transcript_Save_Enabled_Mobile',
```
Comment (Shailesh351) [inline app/livechat/server/config.ts]:
```suggestion
			i18nLabel: 'Transcript_Enabled_Mobile',
```

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 4]
```diff
@@ -4365,9 +4365,10 @@
   "totp-invalid": "Code or password invalid",
   "totp-required": "TOTP Required",
   "Transcript": "Transcript",
-  "Transcript_Enabled": "Ask Visitor if They Would Like a Transcript After Chat Closed",
+  "Transcript_Enabled": "Enable Save Transcript",
   "Transcript_message": "Message to Show When Asking About Transcript",
   "Transcript_Print_Enabled": "Allow Visitor to Print the Transcript",
+  "Transcript_Save_Enabled_Mobile": "Allow Visitor to Print the Transcript on Mobile",
```
Comment (Shailesh351) [inline packages/rocketchat-i18n/i18n/en.i18n.json]:
```suggestion
  "Transcript_Enabled": "Enable Save Transcript",
  "Transcript_Enabled_Mobile": "Allow Visitor to Print the Transcript on Mobile",
  "Transcript_message": "Message to Show When Asking About Transcript",
  "Transcript_Print_Enabled": "Allow Visitor to Print the Transcript",
```

**Comment by ear-dev** [comment]
@Shailesh351 is this one good to go?

**Review by Shailesh351 — APPROVED** [review]

**Pull request merged**